commit d96820d46bc315f669789e18d4aef76d3bc71cc0
Author: Kevin Yang <yyd@google.com>
Date:   Wed Aug 22 17:43:15 2018 -0400

    tcp_bbr: in restart from idle, see if we should exit PROBE_RTT
    
    [ Upstream commit 5490b32dce6932ea7ee8e3b2f76db2957c92af6e ]
    
    This patch fix the case where BBR does not exit PROBE_RTT mode when
    it restarts from idle. When BBR restarts from idle and if BBR is in
    PROBE_RTT mode, BBR should check if it's time to exit PROBE_RTT. If
    yes, then BBR should exit PROBE_RTT mode and restore the cwnd to its
    full value.
    
    Fixes: 0f8782ea1497 ("tcp_bbr: add BBR congestion control")
    Signed-off-by: Kevin Yang <yyd@google.com>
    Signed-off-by: Neal Cardwell <ncardwell@google.com>
    Reviewed-by: Yuchung Cheng <ycheng@google.com>
    Reviewed-by: Soheil Hassas Yeganeh <soheil@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/ipv4/tcp_bbr.c b/net/ipv4/tcp_bbr.c
index 5880015dee36..e99d6afb70ef 100644
--- a/net/ipv4/tcp_bbr.c
+++ b/net/ipv4/tcp_bbr.c
@@ -174,6 +174,8 @@ static const u32 bbr_lt_bw_diff = 4000 / 8;
 /* If we estimate we're policed, use lt_bw for this many round trips: */
 static const u32 bbr_lt_bw_max_rtts = 48;
 
+static void bbr_check_probe_rtt_done(struct sock *sk);
+
 /* Do we estimate that STARTUP filled the pipe? */
 static bool bbr_full_bw_reached(const struct sock *sk)
 {
@@ -304,6 +306,8 @@ static void bbr_cwnd_event(struct sock *sk, enum tcp_ca_event event)
 		 */
 		if (bbr->mode == BBR_PROBE_BW)
 			bbr_set_pacing_rate(sk, bbr_bw(sk), BBR_UNIT);
+		else if (bbr->mode == BBR_PROBE_RTT)
+			bbr_check_probe_rtt_done(sk);
 	}
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1279_d96820d46bc315f669789e18d4aef76d3bc71cc0.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1279_d96820d46bc315f669789e18d4aef76d3bc71cc0.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
